pkg/apis/stable/v1alpha1/fleet.go

```diff
@@ -134,6 +137,24 @@ func (f *Fleet) ApplyDefaults() {
 	f.ObjectMeta.Annotations[stable.VersionAnnotation] = pkg.Version
 }
 
+// Validate validates the Fleet configuration.
+// If a Fleet is invalid there will be > 0 values in
+// the returned array
+func (f *Fleet) Validate() (bool, []metav1.StatusCause) {
```

Comment:
Something I'm fixing elsewhere, for the other validate functions (PR incoming) - I think we should file the return value, so the return should be: `([]metav1.StatusCause, bool)` - just because that's what we do in go for everything else.

Check a map:  `v, ok :=  map[key]` for example. So we should order it the same.

I've got the rest in an upcoming PR that also has new Validate functions as well.

How do you feel about that?

Reply:
yes, that would be much more natural.